platform/jvm/capture/src/test/kotlin/io/bitdrift/capture/ProvidersTest.kt

```diff
@@ -7,120 +7,22 @@
 class ProvidersTest {
+
     @Test
     fun system_date_provider() {
-        // ACT - just exercise the codepath to make sure we're not using invalid APIs for version 23
-        val provider = SystemDateProvider()
-        val date = provider()
+        // Just exercise the codepath to make sure we're not using invalid APIs for version 23
+        val date = SystemDateProvider().invoke()
 
-        // ASSERT
         assertThat(date).isNotNull
     }
-
-    @Test
-    fun client_attributes() {
-        // ARRANGE
-        val packageName = "my.bitdrift.test"
-        val versionName = "1.2.3.4"
-        val versionCode = 66
-
-        val currentState = Lifecycle.State.STARTED
-        val lifecycle = mock(Lifecycle::class.java)
-        doReturn(currentState).`when`(lifecycle).currentState
-        val lifecycleOwner = mock(LifecycleOwner::class.java)
-        doReturn(lifecycle).`when`(lifecycleOwner).lifecycle
-
-        val context = spy(ApplicationProvider.getApplicationContext<Context>())
-        doReturn(packageName).`when`(context).packageName
-
-        val packageManager: PackageManager = mock(PackageManager::class.java)
-        doReturn(packageManager).`when`(context).packageManager
-
-        val packageInfo: PackageInfo = mock(PackageInfo::class.java)
-        doReturn(packageInfo).`when`(packageManager).getPackageInfo(packageName, 0)
-        packageInfo.versionName = versionName
-        packageInfo.versionCode = versionCode
-
-        // ACT
-        val clientAttributes = ClientAttributes(context, lifecycleOwner).invoke()
-
-        // ASSERT
-        assertThat(clientAttributes).containsEntry("app_version", versionName)
-        assertThat(clientAttributes).containsEntry("_app_version_code", versionCode.toString())
-        assertThat(clientAttributes).containsEntry("app_id", packageName)
-        assertThat(clientAttributes).containsEntry("foreground", "1")
-    }
-
-    @Test
-    fun client_attributes_not_foreground() {
-        // ARRANGE
-        val currentState = Lifecycle.State.CREATED
-        val lifecycle = mock(Lifecycle::class.java)
-        doReturn(currentState).`when`(lifecycle).currentState
-        val lifecycleOwner = mock(LifecycleOwner::class.java)
-        doReturn(lifecycle).`when`(lifecycleOwner).lifecycle
-
-        val context = spy(ApplicationProvider.getApplicationContext<Context>())
-
-        val packageManager: PackageManager = mock(PackageManager::class.java)
-        doReturn(packageManager).`when`(context).packageManager
-
-        val packageInfo: PackageInfo = mock(PackageInfo::class.java)
-        doReturn(packageInfo).`when`(packageManager).getPackageInfo("my.bitdrift.test", 0)
-
-        // ACT
-        val clientAttributes = ClientAttributes(context, lifecycleOwner).invoke()
-
-        // ASSERT
-        assertThat(clientAttributes).containsEntry("foreground", "0")
-    }
-
-    @Test
-    @Ignore("Robolectric throwing ClassNotFoundException: android.telephony.TelephonyCallback")
```

Comment:
what was the actual fix of the TODO/failure? Was it just bumping Robolectric to 4.13?

Reply:
That's correct. That issue was fixed in https://github.com/robolectric/robolectric/pull/7786 and released as part of [Robolectric `4.10`](https://github.com/robolectric/robolectric/releases/tag/robolectric-4.10) - it seems that the test was marked as ignored at some point and never revisited ¯\\\_(ツ)\_/¯

That being said, I bumped to the latest available to avoid `ClientAttributesTest#package_info_android_tiramisu()`'s issue `Skip package_info_android_tiramisu because Robolectric doesn't support legacy resources mode after P` (which gets fixed using Robolectric 4.11+).